starter-code/scripts/articleView.js

```diff
@@ -55,15 +62,39 @@ articleView.handleCategoryFilter = function() {
   //       When an option with a value is selected, hide all the articles, then reveal the matches.
   //       When the blank (default) option is selected, show all the articles, except for the template.
   //       Be sure to reset the #author-filter while you are at it!
-
-};
+  // Estimated 20 mins, It actually took : 30 mins
+  $('#category-filter').on('change', function() {
+    var category = $(this).val()
+    if (category) {
+      $('article').fadeOut();
+      var selector = 'article[data-category="'+ category +'"]'
+      $(selector).fadeIn();
+    } else {
+      $('article:not(.template)').fadeIn();
+    }
+  })
+}
 
 articleView.handleMainNav = function() {
   // TODO: Add an event handler to .main-nav elements that will power the Tabs feature.
   //       Clicking any .tab element should hide all the .tab-content sections, and then reveal the
   //       single .tab-content section that is associated with the clicked .tab element.
   //       So: You need to dynamically build a selector string with the correct ID, based on the
   //       data available to you on the .tab element that was clicked.
+  //  Estimated time: 15 mins, It actually took: 25 mins
+  $('.tab').on('click', function(){
```

Comment:
This is working well, but I want to challenge you :) . You have this code on 90/91 and 94/95. My challenge is to condense it to one line! You can also condense the if/else if statement. 
Hint: Instead of targeting specific ids/classes, could we grab the one targeted and do our code on it?
You are not losing points for this, just wanted to add a challenge and an opportunity to DRY out your code!